spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/ContractsCopyTask.java

```diff
@@ -0,0 +1,455 @@
+import org.eclipse.jgit.lib.Ref;
+import org.eclipse.jgit.transport.UsernamePasswordCredentialsProvider;
+import org.gradle.api.Action;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.GradleException;
+import org.gradle.api.Task;
+import org.gradle.api.file.DirectoryProperty;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.MapProperty;
+import org.gradle.api.provider.Property;
+import org.gradle.api.tasks.CacheableTask;
+import org.gradle.api.tasks.Input;
+import org.gradle.api.tasks.InputDirectory;
+import org.gradle.api.tasks.Internal;
+import org.gradle.api.tasks.Nested;
+import org.gradle.api.tasks.Optional;
+import org.gradle.api.tasks.OutputDirectory;
+import org.gradle.api.tasks.PathSensitive;
+import org.gradle.api.tasks.PathSensitivity;
+import org.gradle.api.tasks.TaskAction;
+import org.springframework.cloud.contract.stubrunner.ContractDownloader;
+import org.springframework.cloud.contract.stubrunner.ScmStubDownloaderBuilder;
+import org.springframework.cloud.contract.stubrunner.StubConfiguration;
+import org.springframework.cloud.contract.stubrunner.StubDownloader;
+import org.springframework.cloud.contract.stubrunner.StubDownloaderBuilderProvider;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptions;
+import org.springframework.cloud.contract.stubrunner.StubRunnerOptionsBuilder;
+import org.springframework.cloud.contract.stubrunner.StubRunnerPropertyUtils;
+import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
+import org.springframework.cloud.contract.verifier.converter.ToYamlConverter;
+import org.springframework.util.StringUtils;
+
+// TODO: Convert to incremental task: https://docs.gradle.org/current/userguide/custom_tasks.html#incremental_tasks
+/**
+ * Task that copies the contracts in order for the jar task to
+ * generate the jar. It takes into consideration the inclusion
+ * patterns when working with repo with shared contracts.
+ *
+ * @author Marcin Grzejszczak
+ * @author Anatoliy Balakirev
+ * @author Shannon Pamperl
+ * @since 1.0.2
+ */
+@CacheableTask
+class ContractsCopyTask extends DefaultTask {
+
+	static final String TASK_NAME = "copyContracts";
+	static final String CONTRACTS = "contracts";
+	static final String BACKUP = "original";
+
+	// inputs
+	private final Property<Boolean> convertToYaml;
+
+	private final Property<Boolean> failOnNoContracts;
+
+	private final DirectoryProperty contractsDirectory;
+
+	private final Dependency contractDependency;
+
+	private final Repository contractRepository;
+
+	private final Property<StubRunnerProperties.StubsMode> contractsMode;
+
+	private final MapProperty<String, String> contractsProperties;
+
+	private final Property<String> contractsPath;
+
+	// FIXME: I don't think this makes sense given the default configuration... This would only make sense if contractDir == project.projectDir
+	private final Property<Boolean> excludeBuildFolders;
```

Comment:
AFAIR this one was related to having a single repository with all the contracts. I'd have to double check what was the exact reason for adding this. I think that when building a JAR with all contracts the build folder would be also automatically attached and put in the stubs jar.